src/myplex.ts

```diff
@@ -20,6 +20,94 @@ import { MediaContainer } from './util.js';
+   * to fetch a token and potentially compromise username and password. To use first call `getWebLogin()`
+   * and present the returned uri to a user to go to, then await `webLoginCheck()`. If you pass in a
+   * `forwardUrl`, then send the user to the returned uri, and when a request comes in on the passed in
+   * url, then await `webLoginCheck()`.
+   */
+  static async getWebLogin(forwardUrl: string | null = null): Promise<WebLogin> {
+    const appName = BASE_HEADERS['X-Plex-Product'];
+    const clientIdentifier = BASE_HEADERS['X-Plex-Client-Identifier'];
+    const pin = await ofetch<Omit<WebLogin, 'uri'>>('https://plex.tv/api/v2/pins', {
+      method: 'post',
+      headers: {
+        Accept: 'application/json'
+      },
+      query: {
+        strong: 'true',
+        'X-Plex-Product': appName,
+        'X-Plex-Client-Identifier': clientIdentifier,
+      },
+    });
+    return {...pin,
+      uri: `https://app.plex.tv/auth#?clientID=${
+        encodeURIComponent(clientIdentifier)
+      }&code=${
+        encodeURIComponent(pin.code)
+      }&context%5Bdevice%5D%5Bproduct%5D=${
+        encodeURIComponent(appName)
+      }${forwardUrl ?
+        '&forwardUrl=' + encodeURIComponent(forwardUrl) :
+        ''
+      }`
+    };
+  }
+
+  /**
+   * Pass in the `webLogin` object obtained from `getWebLogin()` and this will poll Plex to see if
+   * the user agreed. It returns a connected `MyPlexAccount` or throws an error.
+   */
+  static async webLoginCheck(webLogin: WebLogin, timeoutSeconds = 60): Promise<MyPlexAccount> {
+    const recheckMs = 3000;
+    const clientIdentifier = BASE_HEADERS['X-Plex-Client-Identifier'];
+    const uri = `https://plex.tv/api/v2/pins/${webLogin.id}`;
+    const startTime = Date.now();
+    while (Date.now() < startTime + (timeoutSeconds * 1000)) {
+      try {
+        // eslint-disable-next-line no-await-in-loop
+        const tokenResponse = await ofetch(uri, {
+          method: 'GET',
+          headers: {
+            Accept: 'application/json',
+          },
+          query: {
+            code: webLogin.code,
+            'X-Plex-Client-Identifier': clientIdentifier,
+          },
+          timeout: recheckMs,
+          retry: 5,
+          retryDelay: recheckMs,
+        });
+        if (tokenResponse.authToken) {
+          const myPlexAccount = new MyPlexAccount(null, '', '', tokenResponse.authToken);
+          // eslint-disable-next-line no-await-in-loop
+          return await myPlexAccount.connect();
+        }
+
+        console.error('sleeping');
+
+        // eslint-disable-next-line no-await-in-loop
+        await sleep(recheckMs);
+      } catch (err) {
+        if ((err as Error).message.includes('aborted')) {
+          continue;
+        }
+
+        if ((err as Error).message.includes('429')) {
+          // eslint-disable-next-line no-await-in-loop
+          await sleep(recheckMs);
+          continue;
+        }
```

Comment:
i think this sort of thing should already be handled by ofetch's retry? https://github.com/unjs/ofetch#%EF%B8%8F-auto-retry

Reply:
oh yeah, I did do that there, afterwards